[Flatpak SDK] Update to FDO 22.08 release

philn opened the pull request:
#### 8b347498dc795dbb61738a9183d6adb90bcc7d2e
<pre>
[Flatpak SDK] Update to FDO 22.08 release
<a href="https://bugs.webkit.org/show_bug.cgi?id=244775">https://bugs.webkit.org/show_bug.cgi?id=244775</a>

Reviewed by Michael Catanzaro.

This new SDK version notably ships:

- ICU 71.1
- GCC 12.1.0
- clang 14.0.6
- Mesa 22.1.7
- Ruby 3.1.2p20
- Python 3.10.6

The libgcrypt downstream patch was dropped, it is not visibly affecting WebCrypto tests. Downstream
Apache patches were removed as well, apr-util was removed and we now use the unified apr instead.
The cmake-lsp recipe was removed, it no longer builds and actually was not very useful anyway. The
libnotify recipe was removed, this is no longer a WebKit dependency. Various recipes we used to
maintain are now used from the FDO SDK, the exception being GTK3, because the upstream version
triggers warnings related with drag-n-drop icons loading.

* Tools/buildstream/Pipfile:
* Tools/buildstream/Pipfile.lock:
* Tools/buildstream/README.md: Added.
* Tools/buildstream/README.rst: Removed.
* Tools/buildstream/elements/flatpak/platform.bst:
* Tools/buildstream/elements/flatpak/sdk.bst:
* Tools/buildstream/elements/freedesktop-sdk.bst:
* Tools/buildstream/elements/sdk-platform.bst:
* Tools/buildstream/elements/sdk/apitrace.bst:
* Tools/buildstream/elements/sdk/capnp.bst:
* Tools/buildstream/elements/sdk/cargo-c.bst:
* Tools/buildstream/elements/sdk/ccls.bst:
* Tools/buildstream/elements/sdk/clangd.bst:
* Tools/buildstream/elements/sdk/cmake-lsp.bst: Removed.
* Tools/buildstream/elements/sdk/enchant-2.bst:
* Tools/buildstream/elements/sdk/ffmpeg.bst:
* Tools/buildstream/elements/sdk/glib-networking.bst: Removed.
* Tools/buildstream/elements/sdk/glib.bst: Removed.
* Tools/buildstream/elements/sdk/graphene.bst:
* Tools/buildstream/elements/sdk/gst-plugin-dav1d.bst:
* Tools/buildstream/elements/sdk/gtk+-3.bst:
* Tools/buildstream/elements/sdk/gtk.bst:
* Tools/buildstream/elements/sdk/icecc.bst:
* Tools/buildstream/elements/sdk/libavif.bst:
* Tools/buildstream/elements/sdk/libcap-ng.bst: Removed.
* Tools/buildstream/elements/sdk/libgcrypt.bst: Removed.
* Tools/buildstream/elements/sdk/libmanette.bst:
* Tools/buildstream/elements/sdk/libnotify.bst: Removed.
* Tools/buildstream/elements/sdk/libsecret.bst:
* Tools/buildstream/elements/sdk/libsoup3.bst: Removed.
* Tools/buildstream/elements/sdk/monado.bst:
* Tools/buildstream/elements/sdk/pygls.bst: Removed.
* Tools/buildstream/elements/sdk/rr.bst:
* Tools/buildstream/elements/sdk/sccache.bst:
* Tools/buildstream/elements/sdk/sparkle-cdm.bst:
* Tools/buildstream/elements/sdk/wpebackend-fdo.bst:
* Tools/buildstream/elements/sdk/xdg-dbus-proxy.bst:
* Tools/buildstream/elements/test-infra.bst:
* Tools/buildstream/elements/test-infra/apr-util.bst: Removed.
* Tools/buildstream/elements/test-infra/apr.bst:
* Tools/buildstream/elements/test-infra/httpd.bst:
* Tools/buildstream/files/gst-plugin-dav1d/Cargo.lock:
* Tools/buildstream/patches/fdo-0001-Bump-libnice-to-current-git-master-HEAD.patch: Removed.
* Tools/buildstream/patches/fdo-0001-gdk-pixbuf-Remove-invalid-meson-options.patch: Removed.
* Tools/buildstream/patches/fdo-0001-gobject-introspection-Bump-to-1.72.patch: Removed.
* Tools/buildstream/patches/fdo-0001-gst-plugins-bad-Remove-invalid-meson-option.patch: Removed.
* Tools/buildstream/patches/fdo-0001-pango-Bump-to-1.50.4.patch: Removed.
* Tools/buildstream/patches/fdo-0001-pipewire-base-Disable-AEC-module.patch: Added.
* Tools/buildstream/patches/fdo-0001-pipewire-base-Track-master-branch-and-disable-AEC-mo.patch: Removed.
* Tools/buildstream/patches/fdo-0002-meson-Bump-to-1.62.patch: Removed.
* Tools/buildstream/patches/gst-plugin-dav1d-fix-build-for-dav1d-1-0.diff: Added.
* Tools/buildstream/patches/httpd-0001-configure-use-pkg-config-for-PCRE-detection.patch: Removed.
* Tools/buildstream/patches/libgcrypt-0001-Use-pkg-config-to-detect-gpg-error.patch: Removed.
* Tools/buildstream/patches/libgcrypt-use-only-dev-urandom-for-testing.patch: Removed.
* Tools/buildstream/project.conf:

Canonical link: <a href="https://commits.webkit.org/254365@main">https://commits.webkit.org/254365@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/36580331d1ba4cf150b09baf018bb452107fbef2

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/88854 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/69/builds/33419 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/19736 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/98056 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/154541 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/92848 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/64/builds/31925 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/27530 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/81098 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/92677 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/94483 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/68/builds/25334 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/75837 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/25286 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/80206 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/80244 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/34/builds/68250 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/29721 "Built successfully") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/14262 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/29451 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/15264 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/32889 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/38197 "Passed tests") | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/75/builds/1281 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/31577 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/34369 "Passed tests") | | 
<!--EWS-Status-Bubble-End-->

mcatanzaro reviewed (approved):
What could go wrong? ;)

philn committed:
[Flatpak SDK] Update to FDO 22.08 release
https://bugs.webkit.org/show_bug.cgi?id=244775

Reviewed by Michael Catanzaro.

This new SDK version notably ships:

- ICU 71.1
- GCC 12.1.0
- clang 14.0.6
- Mesa 22.1.7
- Ruby 3.1.2p20
- Python 3.10.6

The libgcrypt downstream patch was dropped, it is not visibly affecting WebCrypto tests. Downstream
Apache patches were removed as well, apr-util was removed and we now use the unified apr instead.
The cmake-lsp recipe was removed, it no longer builds and actually was not very useful anyway. The
libnotify recipe was removed, this is no longer a WebKit dependency. Various recipes we used to
maintain are now used from the FDO SDK, the exception being GTK3, because the upstream version
triggers warnings related with drag-n-drop icons loading.

* Tools/buildstream/Pipfile:
* Tools/buildstream/Pipfile.lock:
* Tools/buildstream/README.md: Added.
* Tools/buildstream/README.rst: Removed.
* Tools/buildstream/elements/flatpak/platform.bst:
* Tools/buildstream/elements/flatpak/sdk.bst:
* Tools/buildstream/elements/freedesktop-sdk.bst:
* Tools/buildstream/elements/sdk-platform.bst:
* Tools/buildstream/elements/sdk/apitrace.bst:
* Tools/buildstream/elements/sdk/capnp.bst:
* Tools/buildstream/elements/sdk/cargo-c.bst:
* Tools/buildstream/elements/sdk/ccls.bst:
* Tools/buildstream/elements/sdk/clangd.bst:
* Tools/buildstream/elements/sdk/cmake-lsp.bst: Removed.
* Tools/buildstream/elements/sdk/enchant-2.bst:
* Tools/buildstream/elements/sdk/ffmpeg.bst:
* Tools/buildstream/elements/sdk/glib-networking.bst: Removed.
* Tools/buildstream/elements/sdk/glib.bst: Removed.
* Tools/buildstream/elements/sdk/graphene.bst:
* Tools/buildstream/elements/sdk/gst-plugin-dav1d.bst:
* Tools/buildstream/elements/sdk/gtk+-3.bst:
* Tools/buildstream/elements/sdk/gtk.bst:
* Tools/buildstream/elements/sdk/icecc.bst:
* Tools/buildstream/elements/sdk/libavif.bst:
* Tools/buildstream/elements/sdk/libcap-ng.bst: Removed.
* Tools/buildstream/elements/sdk/libgcrypt.bst: Removed.
* Tools/buildstream/elements/sdk/libmanette.bst:
* Tools/buildstream/elements/sdk/libnotify.bst: Removed.
* Tools/buildstream/elements/sdk/libsecret.bst:
* Tools/buildstream/elements/sdk/libsoup3.bst: Removed.
* Tools/buildstream/elements/sdk/monado.bst:
* Tools/buildstream/elements/sdk/pygls.bst: Removed.
* Tools/buildstream/elements/sdk/rr.bst:
* Tools/buildstream/elements/sdk/sccache.bst:
* Tools/buildstream/elements/sdk/sparkle-cdm.bst:
* Tools/buildstream/elements/sdk/wpebackend-fdo.bst:
* Tools/buildstream/elements/sdk/xdg-dbus-proxy.bst:
* Tools/buildstream/elements/test-infra.bst:
* Tools/buildstream/elements/test-infra/apr-util.bst: Removed.
* Tools/buildstream/elements/test-infra/apr.bst:
* Tools/buildstream/elements/test-infra/httpd.bst:
* Tools/buildstream/files/gst-plugin-dav1d/Cargo.lock:
* Tools/buildstream/patches/fdo-0001-Bump-libnice-to-current-git-master-HEAD.patch: Removed.
* Tools/buildstream/patches/fdo-0001-gdk-pixbuf-Remove-invalid-meson-options.patch: Removed.
* Tools/buildstream/patches/fdo-0001-gobject-introspection-Bump-to-1.72.patch: Removed.
* Tools/buildstream/patches/fdo-0001-gst-plugins-bad-Remove-invalid-meson-option.patch: Removed.
* Tools/buildstream/patches/fdo-0001-pango-Bump-to-1.50.4.patch: Removed.
* Tools/buildstream/patches/fdo-0001-pipewire-base-Disable-AEC-module.patch: Added.
* Tools/buildstream/patches/fdo-0001-pipewire-base-Track-master-branch-and-disable-AEC-mo.patch: Removed.
* Tools/buildstream/patches/fdo-0002-meson-Bump-to-1.62.patch: Removed.
* Tools/buildstream/patches/gst-plugin-dav1d-fix-build-for-dav1d-1-0.diff: Added.
* Tools/buildstream/patches/httpd-0001-configure-use-pkg-config-for-PCRE-detection.patch: Removed.
* Tools/buildstream/patches/libgcrypt-0001-Use-pkg-config-to-detect-gpg-error.patch: Removed.
* Tools/buildstream/patches/libgcrypt-use-only-dev-urandom-for-testing.patch: Removed.
* Tools/buildstream/project.conf:

Canonical link: https://commits.webkit.org/254365@main
